vm/src/frame.rs

```diff
@@ -90,7 +90,13 @@ impl fmt::Debug for Scope {
 }
 
 impl Scope {
-    pub fn new(locals: Option<PyDictRef>, globals: PyDictRef) -> Scope {
+    pub fn new(locals: Option<PyDictRef>, globals: PyDictRef, vm: &VirtualMachine) -> Scope {
+        if !globals.contains_key("__builtins__", vm) {
```

Comment:
I would not set the builtins in this new function, but in the calling sites of this new function, as @cthulahoops suggested.

Reply:
I thought @cthulahoops said `__builtins__` is added in every case.

If we still need to keep the `new` clean, I will add `Scope::with_builtins`. Is this ok?